modules/creating-cluster-with-aws-kms-key.adoc

```diff
@@ -0,0 +1,195 @@
+// Module included in the following assemblies:
+//
+// * monitoring/enabling-monitoring-for-user-defined-projects.adoc
+
+:_content-type: PROCEDURE
+[id="creating-cluster-with-aws-kms-key"]
+= Creating a ROSA cluster using custom AWS KMS key
+
+You can create a {product-title} (ROSA) cluster with a customer-provided KMS key that will be used to encrypt both the root volumes of nodes as well as persistent volumes for mounted EBS claims.
+
+.Procedure
+
+. Create a custom AWS customer-managed KMS key:
++
+[source,terminal]
+----
+$ KMS_ARN=$(aws kms create-key --region $AWS_REGION --description 'Custom ROSA Encryption Key' --tags TagKey=red-hat,TagValue=true --query KeyMetadata.Arn --output text)
+----
++
+This command will save the ARN output of this custom key for further steps.
+
+. Verify the KMS key has been created:
++
+[source,terminal]
+----
+$ echo $KMS_ARN
+----
+
+. Generate the necessary key policy to allow the ROSA STS roles to access the custom KMS key. Use the below command to populate a sample policy, or create your own.
+
++
+[source,terminal]
+----
+$ EBS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-openshift-cluster-csi-drivers-ebs-cloud-credential" | cut -c -64) 
+KUBE_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kube-controller-manager" | cut -c -64)
+KMS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kms-provider" | cut -c -64)
+CAPA_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-capa-controller-manager" | cut -c -64) <1>
+
+cat << EOF > rosa-key-policy.json
+{
```

Comment:
Per doc guidelines: [Do not use more than one command per code block.](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#single-command-per-code-block)

Reply:
separated into 2 blocks, thanks

Reply:
Similar to above feedback, were referencing a bunch of environment variables that haven't been set. That's not a _bad_ thing, to use environment variables IMO...we should just revisit the existing modules to include a step that sets these values to environment variables